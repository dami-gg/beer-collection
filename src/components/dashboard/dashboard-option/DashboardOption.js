// @flow

import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class DashboardOption extends Component {
  props:{
    url: string,
    image: string,
    title: string
  };

  render() {
    return (
        <div className="dashboard__option">
          <Link to={this.props.url}>
            <img src={this.props.image} className="dashboard__option__image" alt="logo"/>
            <p className="dashboard__option__title">{this.props.title}</p>
          </Link>
        </div>
    );
  }
}

export default DashboardOption;
