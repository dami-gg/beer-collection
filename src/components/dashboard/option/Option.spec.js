import React from 'react';
import DashboardOption from './Option';
import {Link} from 'react-router';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('DashboardOption', () => {
  let component;
  let elements;
  let mock;

  it('should render without crashing', () => {
    shallow(<DashboardOption />);
  });

  it('should match the snapshot', () => {
    component = renderer.create(<DashboardOption />);
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('render', () => {
    beforeEach(setup);

    it('should render a Link component', () => {
      expect(component.find(Link).length).toEqual(1);
    });

    it('should have a link to the url specified in the props', () => {
      elements = component.find(Link);

      expect(elements.first().props().to).toEqual(mock.props.url);
    });

    it('should render an image with the source specified in the props', () => {
      elements = component.find('.dashboard__option__image');

      expect(elements.first().props().src).toEqual(mock.props.image);
    });

    it('should render a title with the value specified in the props', () => {
      elements = component.find('.dashboard__option__title');

      expect(elements.first().text()).toEqual(mock.props.title);
    });
  });

  function setup() {
    mocks();
    render();
  }

  function render() {
    component = shallow(
        <DashboardOption
          title={mock.props.title}
          url={mock.props.url}
          image={mock.props.image}
        />
    );
  }

  function mocks() {
    mock = {
      props: {
        url: 'URL',
        image: 'IMAGE',
        title: 'TITLE'
      }
    };
  }
});

