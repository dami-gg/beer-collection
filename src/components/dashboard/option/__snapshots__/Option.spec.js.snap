// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should match the snapshot 1`] = `
<div
  className="dashboard__option"
>
  <a
    className="dashboard__option__card"
    href="/url"
    onClick={[Function]}
  >
    <div
      className="dashboard__option__image-wrapper"
    >
      <img
        alt="logo"
        className="dashboard__option__image"
        src="IMAGE"
      />
    </div>
    <div
      className="dashboard__option__title-wrapper"
    >
      <p
        className="dashboard__option__title"
      >
        TITLE
      </p>
    </div>
  </a>
</div>
`;
